src/linter.ts

```diff
@@ -158,74 +145,78 @@ class Linter {
         };
     }
 
-    // Applies fixes to the files where the failures are reported.
-    // Returns the content of the source file which AST needs to be reloaded.
-    protected applyFixes(sourceFilePath: string, sourceContent: string, ruleFailures: RuleFailure[]) {
-      const fixesPerFile: {[file: string]: Fix[]} = ruleFailures
-          .reduce((accum: {[file: string]: Fix[]}, c) => {
-              const currentFileName = c.getFileName();
-              const fix = c.getFix();
-              if (fix) {
-                  accum[currentFileName] = accum[currentFileName] || [];
-                  accum[currentFileName].push(fix);
-              }
-              return accum;
-          }, {});
-
-      const hasFixes = Object.keys(fixesPerFile).length > 0;
-      let result = sourceContent;
-
-      if (hasFixes) {
-          this.fixes = this.fixes.concat(ruleFailures);
-          Object.keys(fixesPerFile).forEach((currentFileName: string) => {
-              const fixesForFile = fixesPerFile[currentFileName];
-              let source = fs.readFileSync(currentFileName, { encoding: "utf-8" });
-              source = Replacement.applyFixes(source, fixesForFile);
-              fs.writeFileSync(currentFileName, source, { encoding: "utf-8" });
-              if (sourceFilePath === currentFileName) {
-                  result = source;
-              }
-          });
-      }
-      return result;
+    private getAllFailures(sourceFile: ts.SourceFile, enabledRules: IRule[]): RuleFailure[] {
+        const failures = flatMap(enabledRules, (rule) => this.applyRule(rule, sourceFile));
+        return removeDisabledFailures(sourceFile, failures);
+    }
+
+    private applyAllFixes(
+            enabledRules: IRule[], fileFailures: RuleFailure[], sourceFile: ts.SourceFile, sourceFileName: string): RuleFailure[] {
+        // When fixing, we need to be careful as a fix in one rule may affect other rules.
+        // So fix each rule separately.
+        let source: string = sourceFile.text;
+
+        for (const rule of enabledRules) {
+            const hasFixes = fileFailures.some((f) => f.hasFix() && f.getRuleName() === rule.getOptions().ruleName);
```

Comment:
Well, the way the enable/disable logic is done here, it's better to handle disables for all rules at once. Otherwise we would have to parse disables after every rule.